assignment/week4/src/types/api/auth.ts

```diff
@@ -0,0 +1,20 @@
+export type SignInRequest = {
+  loginId: string;
+  password: string;
+};
+
+export type SignInResponse = {
+  data: {
+    userId: string;
+  };
+};
+
+export type SignUpRequest = {
+  loginId: string;
+  password: string;
+  nickname: string;
+};
+
+export type SignUpResponse = {
+  nickname: string;
+};
```

Comment:
```suggestion
type UserInfo = {
  loginId: string;
  password: string;
  nickname: string;
  userId: string;
};

export type SignInRequest = Pick<UserInfo, "loginId" | "password">;

export type SignUpRequest = Omit<UserInfo, "userId">;

export type SignInResponse = {
  data: Pick<UserInfo, "userId">;
};

export type SignUpResponse = Pick<UserInfo, "nickname">;
```

이렇게 유틸리티 타입을 사용해서 변경에 유연한 타입을 만들 수도 있어요.

Reply:
이렇게 접근을 할 수 있었네요... 저도 하나 배워갑니다 👍